sycl/include/CL/sycl/detail/queue_impl.hpp

```diff
@@ -132,18 +135,20 @@ class queue_impl {
       CreationFlags |= CL_QUEUE_PROFILING_ENABLE;
     }
 
-    cl_int Error = CL_SUCCESS;
-    cl_command_queue Queue;
-    cl_context ClContext = detail::getSyclObjImpl(m_Context)->getHandleRef();
+    RT::PiResult Error = PI_SUCCESS;
+    RT::PiQueue Queue;
+    RT::PiContext Context = detail::getSyclObjImpl(m_Context)->getHandleRef();
+    RT::PiDevice Device = detail::getSyclObjImpl(m_Device)->getHandleRef();
 #ifdef CL_VERSION_2_0
     cl_queue_properties CreationFlagProperties[] = {
         CL_QUEUE_PROPERTIES, CreationFlags, 0};
-    Queue = clCreateCommandQueueWithProperties(
-        ClContext, m_Device.get(), CreationFlagProperties,
-        &Error);
+    PI_CALL((Queue = RT::piQueueCreate(
+        Context, Device, CreationFlagProperties, &Error), Error));
 #else
-    Queue = clCreateCommandQueue(ClContext, m_Device.get(),
-                                          CreationFlags, &Error);
+    // TODO: do we really need this old interface into PI and here?
```

Comment:
No. This should be handled by OpenCL plugin. The plug-in should know where to lower `piQueueCreate` to `clCreateCommandQueueWithProperties` or to `clCreateCommandQueue`.